app/ante/ante.go

```diff
@@ -70,59 +90,41 @@ func NewAnteHandler(ak auth.AccountKeeper, evmKeeper EVMKeeper, sk types.SupplyK
 	}
 }
 
-// sigGasConsumer overrides the DefaultSigVerificationGasConsumer from the x/auth
-// module on the SDK. It doesn't allow ed25519 nor multisig thresholds.
-func sigGasConsumer(
-	meter sdk.GasMeter, _ []byte, pubkey tmcrypto.PubKey, _ types.Params,
+var _ authante.SignatureVerificationGasConsumer = DefaultSigVerificationGasConsumer
+
+// DefaultSigVerificationGasConsumer is the default implementation of SignatureVerificationGasConsumer. It consumes gas
+// for signature verification based upon the public key type. The cost is fetched from the given params and is matched
+// by the concrete type.
+func DefaultSigVerificationGasConsumer(
+	meter sdk.GasMeter, sig signing.SignatureV2, params authtypes.Params,
 ) error {
-	switch pubkey.(type) {
-	case ethsecp256k1.PubKey:
-		meter.ConsumeGas(secp256k1VerifyCost, "ante verify: secp256k1")
+	pubkey := sig.PubKey
+	switch pubkey := pubkey.(type) {
+	case *ed25519.PubKey:
+		meter.ConsumeGas(params.SigVerifyCostED25519, "ante verify: ed25519")
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidPubKey, "ED25519 public keys are unsupported")
```

Comment:
why is it consuming gas if it's an unsupported key?

Reply:
good catch, although this is supported for validators